controller/events.go

```diff
@@ -177,6 +370,13 @@ func (e *EventListener) Notify(event *ct.AppEvent) {
 			sub.Notify(event)
 		}
 	}
+	if event.AppID != "" {
```

Comment:
Not sure I follow. Let's say I subscribe to artifact changes whose events don't have `AppID` set, my subscriber will be in `subscribers[""]`, and when an artifact event comes along, `event.AppID` will be `""` so I won't get notified due to this condition?